src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -296,7 +296,71 @@ asset::SAssetBundle CImageLoaderTGA::loadAsset(io::IReadFile* _file, const asset
     for (auto& img : images)
         img->drop();
     return SAssetBundle({core::smart_refctd_ptr<IAsset>(tex, core::dont_grab)});
+
+
+
 #else
+	core::vector<asset::ICPUImage*> images;
+
+	switch (header.ImageType)
+	{
+	case 1: // uncompressed color-mapped image
+	case 2: // uncompressed rgb image
+	case 3: // uncompressed grayscale image
+	case 10:
```

Comment:
well you're falling through your switch statement, so all non-rgba format images will try to load with wrong format.

Reply:
I know, but it is for testing purpose, so far I'm trying to load only an rgba image